corehq/apps/sso/models.py

```diff
@@ -245,6 +313,20 @@ def __str__(self):
         return f"{self.email_domain} authenticated by [{self.identity_provider.name}]"
 
 
+@receiver(pre_save, sender=AuthenticatedEmailDomain)
+@receiver(pre_delete, sender=AuthenticatedEmailDomain)
+def clear_caches_for_email_domain(sender, instance, **kwargs):
```

Comment:
Seems like there is a possibility of a race condition immediately after these are cleared (pre-save/delete) if another request causes the cache to be populated (with soon-to-be-stale data) before the save or delete is actually done. Would it makes sense to use post-save/delete signals instead?

Reply:
updated to `post-save` and `post-delete`